src/components/structures/MatrixChat.js

```diff
@@ -896,6 +897,12 @@ module.exports = React.createClass({
         });
     },
 
+    _setStateFromSessionStore() {
+        this.setState({
+            userHasGeneratedPassword: Boolean(this._sessionStore.getCachedPassword()),
```

Comment:
It's used in `LoggedInView`... I guess I should pass that through explicitly by I then question why `...this.state` is given to the props of `LoggedInView` from `MatrixChat` because of confusion like this.